### ISSUE TITLE
Documentation mismatch for keyword arg on assertWaitFor

### ISSUE THREAD

**Issue by msmcconnell** [ros2/launch#547]
## Bug report

**Required Info:**

- Operating System:
  - 
  Ubuntu 20.04
- Installation type:
  - 
- Version or commit hash:
  - 
  foxy - 5bc6d3403b2067adda2778fbf1bfc8278f9336f9
- DDS implementation:
  - 
  Fast-RTPS
- Client library (if applicable):
  - 
  N/A

#### Steps to reproduce issue

The documentation for assertWaitFor in the launch_testing README.md file (see link)
https://github.com/ros2/launch/blob/5bc6d3403b2067adda2778fbf1bfc8278f9336f9/launch_testing/README.md?plain=1#L147

implies that the user can specify the running process using the ```proc=<my process> ``` keyword arg. However, doing so will result in the following error
```
Traceback (most recent call last):
  File "/workspaces/carma_ws/temp_ws/carma-platform/system_controller/test/launch_test.py", line 86, in test_nominal_lifecycle
    proc_output.assertWaitFor('Initializing SystemControllerNode', proc='test_system_controller', strict_proc_matching=True,timeout=5)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch_testing/io_handler.py", line 145, in assertWaitFor
    success = self.waitFor(*args, **kwargs)
TypeError: waitFor() got an unexpected keyword argument 'proc'
```

This is because the actual keyword is "process" (see link)
https://github.com/ros2/launch/blob/5bc6d3403b2067adda2778fbf1bfc8278f9336f9/launch_testing/launch_testing/io_handler.py#L151

This can be duplicated with any launch test making use of the relevant call for example
```
proc_output.assertWaitFor('Initializing SystemControllerNode', proc='test_system_controller', strict_proc_matching=True,timeout=5)
```
vs
```
proc_output.assertWaitFor('Initializing SystemControllerNode', process='test_system_controller', strict_proc_matching=True,timeout=5)
```
#### Expected behavior
Documentation matches implementation 

#### Actual behavior
Documentation mismatch

#### Additional information

     
----

**Comment by hidmic** [comment]
@msmcconnell a documentation fix would be best. We wouldn't be able to backport an API breakage to Foxy.

**Comment by jacobperron** [comment]
This was fixed for Foxy #554 
I've cherry-picked the same change to Galactic #561 and Rolling #560

**Comment by hidmic** [comment]
Closing as this has been addressed and backported.
